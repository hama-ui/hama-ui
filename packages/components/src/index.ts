export * as button from './components/button'
