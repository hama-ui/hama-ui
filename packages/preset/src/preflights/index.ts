import type { Preflight } from '@unocss/core'
import type { Theme } from '@unocss/preset-mini'

import { preflightReset } from './reset'

export const preflights: Preflight<Theme>[] = [
  preflightReset,
]
