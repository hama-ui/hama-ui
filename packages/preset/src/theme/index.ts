import type { Theme } from '@unocss/preset-mini'

import type { PresetHamaOptions } from '..'

import { boxShadow } from './box-shadow'
import { type PresetHamaThemeColorsOptions, themeColors } from './colors'

export interface PresetHamaThemeOptions {
  colors?: PresetHamaThemeColorsOptions
}

export const theme = (option: PresetHamaThemeOptions = {}, prefix?: PresetHamaOptions['prefix']): Theme => ({
  boxShadow,
  colors: themeColors(option.colors, prefix),
})
