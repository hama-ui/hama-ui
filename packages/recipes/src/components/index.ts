export * as Avatar from './avatar'
export * as Badge from './badge'
export * as Button from './button'
