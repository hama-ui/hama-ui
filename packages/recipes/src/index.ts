export * as Avatar from './components/avatar'
export * as Badge from './components/badge'
export * as Button from './components/button'
