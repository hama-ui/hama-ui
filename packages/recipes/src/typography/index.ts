export * as Heading from './heading'
