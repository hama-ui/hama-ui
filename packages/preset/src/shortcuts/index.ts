import type { UserShortcuts } from '@unocss/core'
import type { Theme } from '@unocss/preset-mini'

import { components } from './components'

export const shortcuts: UserShortcuts<Theme> = [
  components,
]
