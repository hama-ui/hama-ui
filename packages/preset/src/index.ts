import { definePreset } from '@unocss/core'
import { presetAttributify } from '@unocss/preset-attributify'
import { presetMini } from '@unocss/preset-mini'
// https://github.com/unocss/unocss/pull/4053
import presetWebFonts from '@unocss/preset-web-fonts'

import { shortcuts } from './shortcuts'

export interface PresetHamaOptions {
  prefix?: `${string}-`
}

export const presetHama = definePreset((_options: PresetHamaOptions = {}) => ({
  name: '@hama-ui/preset',
  presets: [
    presetMini(),
    presetAttributify(),
    presetWebFonts(),
  ],
  shortcuts,
}))

export default presetHama
