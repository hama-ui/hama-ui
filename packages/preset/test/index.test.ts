import { createGenerator } from '@unocss/core'
import { describe, expect, it } from 'vitest'

import { presetHama } from '../src'

describe('@hama-ui/preset', () => {
  const uno = createGenerator({
    presets: [presetHama()],
  })

  it('preflights', async () => {
    const { css } = await uno.generate('')
    expect(css).toMatchSnapshot()
  })
})
