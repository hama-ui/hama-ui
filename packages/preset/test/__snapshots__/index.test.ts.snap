// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`@hama-ui/preset > preflights 1`] = `
"/* layer: preflights */

*,
::before,
::after {
  box-sizing: border-box; /* 1 */
  border-width: 0; /* 2 */
  border-style: solid; /* 2 */
  border-color: var(--un-default-border-color, #e5e7eb); /* 2 */
}

::before,
::after {
  --un-content: '';
}

html,
:host {
  line-height: 1.5; /* 1 */
  -webkit-text-size-adjust: 100%; /* 2 */
  -moz-tab-size: 4; /* 3 */
  tab-size: 4; /* 3 */
  font-family: ui-sans-serif, system-ui, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji"; /* 4 */
  font-feature-settings: normal; /* 5 */
  font-variation-settings: normal; /* 6 */
  -webkit-tap-highlight-color: transparent; /* 7 */
}

body {
  margin: 0; /* 1 */
  line-height: inherit; /* 2 */
}

hr {
  height: 0; /* 1 */
  color: inherit; /* 2 */
  border-top-width: 1px; /* 3 */
}

abbr:where([title]) {
  text-decoration: underline dotted;
}

h1,
h2,
h3,
h4,
h5,
h6 {
  font-size: inherit;
  font-weight: inherit;
}

a {
  color: inherit;
  text-decoration: inherit;
}

b,
strong {
  font-weight: bolder;
}

code,
kbd,
samp,
pre {
  font-family: ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace; /* 1 */
  font-feature-settings: normal; /* 2 */
  font-variation-settings: normal; /* 3 */
  font-size: 1em; /* 4 */
}

small {
  font-size: 80%;
}

sub,
sup {
  font-size: 75%;
  line-height: 0;
  position: relative;
  vertical-align: baseline;
}

sub {
  bottom: -0.25em;
}

sup {
  top: -0.5em;
}

table {
  text-indent: 0; /* 1 */
  border-color: inherit; /* 2 */
  border-collapse: collapse; /* 3 */
}

button,
input,
optgroup,
select,
textarea {
  font-family: inherit; /* 1 */
  font-feature-settings: inherit; /* 1 */
  font-variation-settings: inherit; /* 1 */
  font-size: 100%; /* 1 */
  font-weight: inherit; /* 1 */
  line-height: inherit; /* 1 */
  color: inherit; /* 1 */
  margin: 0; /* 2 */
  padding: 0; /* 3 */
}

button,
select {
  text-transform: none;
}

button,
[type='button'],
[type='reset'],
[type='submit'] {
  -webkit-appearance: button; /* 1 */
  background-color: transparent; /* 2 */
  background-image: none; /* 2 */
}

:-moz-focusring {
  outline: auto;
}

:-moz-ui-invalid {
  box-shadow: none;
}

progress {
  vertical-align: baseline;
}

::-webkit-inner-spin-button,
::-webkit-outer-spin-button {
  height: auto;
}

[type='search'] {
  -webkit-appearance: textfield; /* 1 */
  outline-offset: -2px; /* 2 */
}

::-webkit-search-decoration {
  -webkit-appearance: none;
}

::-webkit-file-upload-button {
  -webkit-appearance: button; /* 1 */
  font: inherit; /* 2 */
}

summary {
  display: list-item;
}

blockquote,
dl,
dd,
h1,
h2,
h3,
h4,
h5,
h6,
hr,
figure,
p,
pre {
  margin: 0;
}

fieldset {
  margin: 0;
  padding: 0;
}

legend {
  padding: 0;
}

ol,
ul,
menu {
  list-style: none;
  margin: 0;
  padding: 0;
}

dialog {
  padding: 0;
}

textarea {
  resize: vertical;
}

input::placeholder,
textarea::placeholder {
  opacity: 1; /* 1 */
  color: #9ca3af; /* 2 */
}

button,
[role="button"] {
  cursor: pointer;
}

:disabled {
  cursor: default;
}

img,
svg,
video,
canvas,
audio,
iframe,
embed,
object {
  display: block; /* 1 */
  vertical-align: middle; /* 2 */
}

img,
video {
  max-width: 100%;
  height: auto;
}

[hidden] {
  display: none;
}

*,::before,::after{--un-rotate:0;--un-rotate-x:0;--un-rotate-y:0;--un-rotate-z:0;--un-scale-x:1;--un-scale-y:1;--un-scale-z:1;--un-skew-x:0;--un-skew-y:0;--un-translate-x:0;--un-translate-y:0;--un-translate-z:0;--un-ring-offset-shadow:0 0 rgb(0 0 0 / 0);--un-ring-shadow:0 0 rgb(0 0 0 / 0);--un-shadow-inset: ;--un-shadow:0 0 rgb(0 0 0 / 0);--un-ring-inset: ;--un-ring-offset-width:0px;--un-ring-offset-color:#fff;--un-ring-width:0px;--un-ring-color:rgb(147 197 253 / 0.5);}::backdrop{--un-rotate:0;--un-rotate-x:0;--un-rotate-y:0;--un-rotate-z:0;--un-scale-x:1;--un-scale-y:1;--un-scale-z:1;--un-skew-x:0;--un-skew-y:0;--un-translate-x:0;--un-translate-y:0;--un-translate-z:0;--un-ring-offset-shadow:0 0 rgb(0 0 0 / 0);--un-ring-shadow:0 0 rgb(0 0 0 / 0);--un-shadow-inset: ;--un-shadow:0 0 rgb(0 0 0 / 0);--un-ring-inset: ;--un-ring-offset-width:0px;--un-ring-offset-color:#fff;--un-ring-width:0px;--un-ring-color:rgb(147 197 253 / 0.5);}"
`;
